packages/browser-wallet/src/background/identity-issuance.ts

```diff
@@ -12,85 +12,106 @@ import { confirmIdentity } from './confirmation';
-                );
-            });
-            onComplete.then((redirectUrl) => {
-                chrome.tabs.onRemoved.removeListener(closedListener);
-                if (tab.id !== undefined) {
-                    chrome.tabs.remove(tab.id);
-                }
-                respond({
-                    status: BackgroundResponseStatus.Success,
-                    result: redirectUrl.substring(redirectUrl.indexOf(codeUriKey) + codeUriKey.length),
-                });
-            });
-        });
-}
+const respond = async (response: IdentityIssuanceBackgroundResponse) => {
+    // TODO this hack is only needed due to a bug in chrome, which prevents chrome.webRequest hooks from waking up the service worker from inactive state.
+    // Bug will be fixed in chrome v107
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=1024211#c73
+    chrome.alarms.clear(IDP_ALARM_NAME);
 
-async function startIdentityIssuance({
-    baseUrl,
-    ...identityRequestInputs
-}: IdentityRequestInput & { baseUrl: string }) {
     const network = await storedCurrentNetwork.get();
     if (!network) {
         return;
     }
-    const respond = async (response: IdentityIssuanceBackgroundResponse) => {
-        let { status } = response;
-        const pending = await sessionPendingIdentity.get();
-        if (!pending) {
-            status = BackgroundResponseStatus.Aborted;
-        } else {
-            if (response.status === BackgroundResponseStatus.Success) {
-                const newIdentity: PendingIdentity = {
-                    ...pending,
-                    status: CreationStatus.Pending,
-                    location: response.result,
-                };
-                await addIdentity(newIdentity, network.genesisHash);
-                confirmIdentity(newIdentity, network.genesisHash);
-            }
-            await sessionPendingIdentity.remove();
+
+    let { status } = response;
+    const pending = await sessionPendingIdentity.get();
+
+    if (!pending) {
+        status = BackgroundResponseStatus.Aborted;
+    } else {
+        if (response.status === BackgroundResponseStatus.Success) {
+            const newIdentity: PendingIdentity = {
+                ...pending,
+                status: CreationStatus.Pending,
+                location: response.result,
+            };
+            await addIdentity(newIdentity, network.genesisHash);
+            confirmIdentity(newIdentity, network.genesisHash);
+        }
+        await sessionPendingIdentity.remove();
+    }
+
+    await openWindow();
+    bgMessageHandler.sendInternalMessage(InternalMessageType.EndIdentityIssuance, { ...response, status });
+};
+
+export function addIdpListeners() {
+    const closedListener = async (tabId: number) => {
+        const idpTabId = await sessionIdpTab.get();
+
+        if (idpTabId !== undefined && tabId === idpTabId) {
+            chrome.tabs.onRemoved.removeListener(closedListener);
```

Comment:
Now that this is a top-level listener, I don't think it should remove itself? Wouldn't this give an issue if I abort twice in a row?

Reply:
Shouldn't this reset the idpTabId instead? (I guess it only matters if chrome would reuse tab ids)